src/core/class.application.php

```diff
@@ -174,47 +178,68 @@ public function overrideThemeSettings() {
             if ($secondaryColor !== false) {
                 $_SESSION["companysettings.secondarycolor"] = $secondaryColor;
             }
-
-        }else{
-            if(!str_starts_with($_SESSION["companysettings.primarycolor"], "#")){
-                $_SESSION["companysettings.primarycolor"] = "#".$_SESSION["companysettings.primarycolor"];
-                $_SESSION["companysettings.secondarycolor"] = "#".$_SESSION["companysettings.primarycolor"];
+        } else {
+            if (!str_starts_with($_SESSION["companysettings.primarycolor"], "#")) {
+                $_SESSION["companysettings.primarycolor"] = "#" . $_SESSION["companysettings.primarycolor"];
+                $_SESSION["companysettings.secondarycolor"] = "#" . $_SESSION["companysettings.primarycolor"];
             }
         }
 
-        if(isset($_SESSION["companysettings.sitename"]) === false) {
+        if (isset($_SESSION["companysettings.sitename"]) === false) {
             $sitename = $this->settingsRepo->getSetting("companysettings.sitename");
             if ($sitename !== false) {
                 $_SESSION["companysettings.sitename"] = $sitename;
-            }else{
+            } else {
                 $_SESSION["companysettings.sitename"] = $this->config->sitename;
             }
         }
 
-        if(isset($_SESSION["companysettings.language"]) === false || $_SESSION["companysettings.language"] == false) {
+        if (isset($_SESSION["companysettings.language"]) === false || $_SESSION["companysettings.language"] == false) {
             $language = $this->settingsRepo->getSetting("companysettings.language");
             if ($language !== false) {
                 $_SESSION["companysettings.language"] = $language;
-            }else{
+            } else {
                 $_SESSION["companysettings.language"] = $this->config->language;
             }
         }
+    }
+
+    private function checkIfInstalled() {
 
+        if(isset($_SESSION['isInstalled']) && $_SESSION['isInstalled'] === true) {
+            return;
+        }
         //Only run this if the user is not logged in (db should be updated/installed before user login)
-        if($this->login->logged_in()===false) {
+        if($this->auth->logged_in()===false) {
 
-            if($this->settingsRepo->checkIfInstalled() === false && isset($_GET['install']) === false){
-                if(!isset($_GET['act']) || $_GET['act'] != 'api.i18n') {
-                    header("Location:" . BASE_URL . "/install");
-                    exit();
+            if(!isset($_SESSION['isInstalled']) || $_SESSION['isInstalled'] === false) {
```

Comment:
isn't this already covered by the fact that the first if didn't return?

Reply:
Good catch. You are right